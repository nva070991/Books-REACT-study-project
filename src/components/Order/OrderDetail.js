import { useState, React } from 'react'
import MinMax from '../MinMax/MinMax'
import props from './OrderProps'

function OrderDetail(prop) {
  const {
    book,
    callback,
    Delete,
  } = prop
  const [quantity, setBook] = useState(Number(book.quantity))

  const setQuatinty = (qu, id) => {
    setBook(qu)
    callback(String(qu), id)
  }
  return (
    <div style={{
      boxSizing: 'border-box',
      background: 'white',
      height: '300px',
      width: '300px',
      borderRadius: '10px',
      padding: '14px',
      display: 'flex',
      flexDirection: 'column',
    }}
    >
      <div>
        <button
          style={{
            background: 'red', height: '20px', width: '20px', borderRadius: '10px', border: 'none',
          }}
          type="button"
          className="buttons"
          onClick={() => Delete(book.id)}
        >
          X
        </button>

      </div>
      <div>

        <h3 data-testid="h3" style={{ height: '50px' }}>{book.title}</h3>

      </div>
      <div>
        Price
        {' '}
        {Number(book.price)}
        {' '}
        /
        {' '}
        {Number(book.price) * Number(quantity)}
      </div>
      <div style={{ marginTop: '25px' }}>
        Quanity
        {' '}
        {Number(book.quantity)}
      </div>
      <div>
        <a href={Number(book.id)}>Readmore</a>
      </div>
      <div style={{ marginTop: '25px' }}>
        <MinMax
          max={Number(book.rest)}
          current={Number(book.quantity)}
          onChange={(qu) => setQuatinty(qu, Number(book.id))}
        />
      </div>

    </div>
  )
}

OrderDetail.propTypes = props

export default OrderDetail
