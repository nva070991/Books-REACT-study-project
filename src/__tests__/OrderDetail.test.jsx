import { render, screen } from '@testing-library/react'
import React from 'react'
import OrderDetail from '../components/Order/OrderDetail'
import setQuatinty from '../components/Order/Order'

describe('OrderDetail component', () => {
  it('should render OrderDetail component', () => {
    const view = render(
      <OrderDetail
        book={{
          id: '101',
          title: 'Война и мир - Л.Н.Толстой',
          price: '800',
          rest: '10',
          quantity: '1',
        }}
        callback={setQuatinty}
      />,
    )
    expect(view).toMatchSnapshot()
  })

  it('should render Title', () => {
    render(<OrderDetail
      book={{
        id: '101',
        title: 'Война и мир - Л.Н.Толстой',
        price: '800',
        rest: '10',
        quantity: '1',
      }}
      callback={setQuatinty}
    />)
    expect(screen.getByTestId('h3')).toHaveTextContent('Война и мир - Л.Н.Толстой')
  })
})
