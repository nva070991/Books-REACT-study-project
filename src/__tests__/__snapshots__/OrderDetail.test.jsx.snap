// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDetail component should render OrderDetail component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="box-sizing: border-box; background: white; height: 300px; width: 300px; border-radius: 10px; padding: 14px; display: flex; flex-direction: column;"
      >
        <div>
          <button
            class="buttons"
            style="background: red; height: 20px; width: 20px; border-radius: 10px;"
            type="button"
          >
            X
          </button>
        </div>
        <div>
          <h3
            data-testid="h3"
            style="height: 50px;"
          >
            Война и мир - Л.Н.Толстой
          </h3>
        </div>
        <div>
          Price
           
          800
           
          /
           
          800
        </div>
        <div
          style="margin-top: 25px;"
        >
          Quanity
           
          1
        </div>
        <div>
          <a
            href="101"
          >
            Readmore
          </a>
        </div>
        <div
          style="margin-top: 25px;"
        >
          <div
            style="display: flex; justify-content: space-between;"
          >
            <button
              data-testid="button-"
              style="background: red; height: 45px; width: 45px; border-radius: 10px;"
              type="button"
            >
              <span
                style="color: white; font-size: 30px;"
              >
                -
              </span>
            </button>
            <input
              data-testid="input"
              type="hidden"
              value="1"
            />
            <button
              data-testid="button+"
              style="background: green; height: 45px; width: 45px; border-radius: 10px;"
              type="button"
            >
              <span
                style="color: white; font-size: 30px;"
              >
                +
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="box-sizing: border-box; background: white; height: 300px; width: 300px; border-radius: 10px; padding: 14px; display: flex; flex-direction: column;"
    >
      <div>
        <button
          class="buttons"
          style="background: red; height: 20px; width: 20px; border-radius: 10px;"
          type="button"
        >
          X
        </button>
      </div>
      <div>
        <h3
          data-testid="h3"
          style="height: 50px;"
        >
          Война и мир - Л.Н.Толстой
        </h3>
      </div>
      <div>
        Price
         
        800
         
        /
         
        800
      </div>
      <div
        style="margin-top: 25px;"
      >
        Quanity
         
        1
      </div>
      <div>
        <a
          href="101"
        >
          Readmore
        </a>
      </div>
      <div
        style="margin-top: 25px;"
      >
        <div
          style="display: flex; justify-content: space-between;"
        >
          <button
            data-testid="button-"
            style="background: red; height: 45px; width: 45px; border-radius: 10px;"
            type="button"
          >
            <span
              style="color: white; font-size: 30px;"
            >
              -
            </span>
          </button>
          <input
            data-testid="input"
            type="hidden"
            value="1"
          />
          <button
            data-testid="button+"
            style="background: green; height: 45px; width: 45px; border-radius: 10px;"
            type="button"
          >
            <span
              style="color: white; font-size: 30px;"
            >
              +
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
